controllers/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -167,6 +167,15 @@ func (r *Reconciler) ensureUserAccount(logger logr.Logger, murAccount toolchainv
 			if !isStatusPresent {
 				counter.IncrementUserAccountCount(logger, murAccount.TargetCluster)
 			}
+			// check the userAccount has actually been created
+			userAccountRetrieved := &toolchainv1alpha1.UserAccount{}
+			if err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+				Namespace: userAccount.Namespace,
+				Name:      userAccount.Name}, userAccountRetrieved); err != nil {
+				if errors.IsNotFound(err) {
+					return updateStatusConditions(logger, r.Client, mur, toBeNotReady(toolchainv1alpha1.MasterUserRecordProvisioningReason, "UserAccount created isn't ready yet"))
+				}
+			}
```

Comment:
Do we need to check if the UserAccount exists here since we're creating it just a few lines above? Are we supposed to check if the UserAccount is Ready instead? That would need to be done outside of this block though, maybe just after the Space status check?